### ISSUE TITLE
startswith first arg must be bytes or a tuple of bytes, not str

### ISSUE THREAD

**Issue by RalphCorderoy** [wking/rss2email#94]
rss2email-wking 3.9-1 on Arch Linux. Running http://www.godbolt.org/feed gives

```
error while running time limited function: startswith first arg must be bytes or a tuple of bytes, not str
cause: startswith first arg must be bytes or a tuple of bytes, not str
```

Here's the current content of the feed. [godbolt-feed.rss.txt](https://github.com/wking/rss2email/files/442034/godbolt-feed.rss.txt)

**Comment by dme** [comment]
In the feedparser _FeedParserMixin class `pop` function, code was added to ensure that `pieces` is a string "for Python 3". Unfortunately this happens too late, as the chunk of code above that removes a `div` wrapper already assumes strings. Moving the code that ensures `pieces` is a string above the `div` removing loop fixes this for me.

(I will go over to the feedparser project and raise an issue there.)

**Comment by dme** [comment]
Pull request for feedparser is here: https://github.com/kurtmckee/feedparser/pull/140

**Comment by Ekleog** [comment]
Hello,

This repository has been deprecated for a few years now, and has been replaced by https://github.com/rss2email/rss2email .

If this issue is still relevant to you, and not fixed with v3.12.2, could you please reopen the issue there?

Cheers,
  Leo

**Comment by RalphCorderoy** [comment]
@dme's above fix to feedparser stopped the error. Closing issue.
